code/script.js

```diff
@@ -222,104 +385,189 @@ const generateBoard = () => {
+      alert(
+        `Yes, the doggie has a ${value} coat! Keep all doggies that have a ${value} coat!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the doggie doesn't have a ${value} coat! Remove all doggies that have a ${value} coat!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
+  } else if (category === 'ears') {
+    if (keep) {
+      alert(
+        `Yes, the doggie has ${value} ears! Keep all doggies that have ${value} ears!`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
+    } else {
+      alert(
+        `No, the doggie doesn't have ${value} ears! Remove all doggies that have ${value} ears!`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+    }
+  }
 
-    for accessories and other
+  if (category === 'nose') {
+    if (keep) {
+      alert(
+        `Yes, the doggie has a ${value} nose! Keep all doggies that have a ${value} nose!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(
+        `No, the doggie doesn't have a ${value} nose! Remove all doggies that have a ${value} nose!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  } else if (category === 'clothing') {
+    if (keep) {
+      alert(
+        `Yes, the doggie wears ${value}! Keep all doggies that wears ${value}!`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    } else {
+      alert(
+        `No, the doggie doesn't wear ${value}! Remove all doggies that wears ${value}!`
+      )
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+    }
+  } else if (category === 'accessories') {
+    if (keep) {
+      alert(
+        `Yes, the doggie has ${value}! Keep all doggies that have ${value}!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(
+        `No, the doggie doesn't have ${value}! Remove all doggies that have ${value}!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  }
+  
+  // Invokes a function to redraw the board with the remaining dogs
+  generateBoard(charactersInPlay)
```

Comment:
As we've been doing the same project I compared to mine, I have just used the 
generateBoard()
does that work for your code?